spectator-api/src/main/java/com/netflix/spectator/api/patterns/PolledGauge.java

```diff
@@ -0,0 +1,402 @@
+ * be to report the number of connections to a {@link com.netflix.spectator.api.DistributionSummary}
+ * every time a connection is added or removed. The distribution summary would provide more
+ * accurate tracking such as max and average number of connections across an interval of time.
+ * The polled gauge would not provide that level of detail.</p>
+ *
+ * <p>If multiple values are monitored with the same id, then the values will be aggregated and
+ * the sum will be reported. For example, registering multiple gauges for active threads in
+ * a thread pool with the same id would produce a value that is the overall number
+ * of active threads. For other behaviors, manage it on the user side and avoid multiple
+ * registrations.</p>
+ */
+public final class PolledGauge {
+
+  private PolledGauge() {
+  }
+
+  /**
+   * Return a builder for configuring a polled gauge reporting to the provided registry.
+   *
+   * @param registry
+   *     Registry that will maintain the state and receive the sampled values for the
+   *     configured gauge.
+   * @return
+   *     Builder for configuring a polled gauge.
+   */
+  public static IdBuilder<Builder> using(Registry registry) {
+    return new IdBuilder<Builder>(registry) {
+      @Override protected Builder createTypeBuilder(Id id) {
+        return new Builder(registry, id);
+      }
+    };
+  }
+
+  /** Force the polling of all gauges associated with the registry. */
+  public static void update(Registry registry) {
+    registry.state().values().forEach(obj -> {
+      if (obj instanceof GaugeTuple) {
+        ((GaugeTuple) obj).doUpdate(registry);
+      }
+    });
+  }
+
+  /**
+   * Explicitly disable polling for the gauge registered with {@code id}. This is optional
+   * and is mostly used if it is desirable for the gauge to go away immediately. The polling
+   * will stop automatically when the referred object is garbage collected. See
+   * {@link Builder#monitor(Object, ToDoubleFunction)} for more information.
+   */
+  public static void remove(Id id) {
+
+  }
+
+  /**
+   * Builder for configuring a polled gauge value.
+   */
+  public static final class Builder extends TagsBuilder<Builder> {
+
+    private final Registry registry;
+    private final Id baseId;
+    private ScheduledExecutorService executor;
+
+    /** Create a new instance. */
+    Builder(Registry registry, Id baseId) {
+      super();
+      this.registry = registry;
+      this.baseId = baseId;
+    }
+
+    /**
+     * Set the executor to be used for polling the value. If not set, then the default
+     * executor will be used which is limited to a single thread to minimize the worst
+     * case resource usage for collecting the gauge data. Use a custom executor if more
+     * resources are needed or if the polling operation is expensive.
+     *
+     * @return
+     *     This builder instance to allow chaining of operations.
+     */
+    public Builder withExecutor(ScheduledExecutorService executor) {
```

Comment:
I went with `scheduleOn` which is similar to the `observeOn` for rxjava Flowable.